### ISSUE TITLE
External control and/or global key bindings

### ISSUE THREAD

**Issue by wm4** [cmdrkotori/mpc-qt#11]
I'd like to control mpc-qt playback via global keyboard shortcuts (at least start/stop/toggle pause and next/previous playlist entry).

I see a few possibilities how this could be implemented:
1. a native feature which grabs global keys
2. a simple way to control a mpc-qt instance from outside (e.g. like mpv's JSON IPC)
3. something more complicated like mpris

I think 2. would probably be best, but I can see the others also being popular with users, especially since this is a GUI application.

**Comment by cmdrkotori** [comment]
Added JSON commands to the local socket at `/tmp/cmdrkotori.mpc-qt` in the form of:
```
{
  "command": "command string"
  "files": [ ... ] (optional, depends on command)
  "file": "/path/to/file" (optional, depends on command)
}
```
The code for parsing these commands lives at `Flow::process_payloadRecieved` in `main.cpp` as of db2b396a. All commands get an ACK if the program hasn't frozen.

Adding MPRIS support would give some sweet desktop integration. Seems a bit wordy though.

**Comment by wm4** [comment]
It works well for me.

Just one thing: I'm missing a command that does
- behave like `start` if nothing is playing
- behave like `pause` if something is playing
- behave like `unpause` if something is playing but in pause state

Also, the `next` and `previous` command could just make the next/previous playlist item selected (without starting playback) if nothing is playing yet.

**Comment by cmdrkotori** [comment]
I have also documented the ipc in [DOCS/ipc.md](https://github.com/cmdrkotori/mpc-qt/blob/master/DOCS/ipc.md).

**Comment by wm4** [comment]
Thanks, the togglePlayback is ideal.
